app/controllers/foreman_deployments/api/v2/deployments_controller.rb

```diff
@@ -0,0 +1,73 @@
+        before_filter :parse_stack, :only => [:show, :configure, :configuration, :run]
+
+        def_param_group :deployment do
+          param :deployment, Hash, :required => true, :action_aware => true do
+            param :name, String, :required => true, :desc => N_('Name for the deployment')
+            param :stack_id, :identifier, :required => true, :desc => N_('Id of the stack to deploy')
+          end
+        end
+
+        api :POST, '/deployments/', N_('Create a deployment')
+        param_group :deployment, :as => :create
+        def create
+          deployment_params = params[:deployment]
+
+          stack = ForemanDeployments::Stack.find(deployment_params.delete(:stack_id)) if deployment_params[:stack_id]
+          deployment_params[:configuration] = ForemanDeployments::Configuration.new(:stack => stack)
+
+          @deployment = ForemanDeployments::Deployment.new(deployment_params)
+          process_response @deployment.save
+        end
+
+        api :GET, '/deployments/', N_('List deployments')
+        def index
+          @deployments = resource_scope
+        end
+
+        api :PUT, '/deployments/:id/configuration/', N_('Configure a deployment')
+        param :id, :identifier, :required => true
+        param :values, Hash, :required => true,
+                             :desc => N_('Hash with configuration for tasks (Format: task ID => configuration values).')
+        def configuration
+          configuration_update = ForemanDeployments::Configuration.new(:values => params[:values])
+
+          ForemanDeployments::Config::LoadVisitor.load(@stack_definition, @deployment.configuration)
+          ForemanDeployments::Config::LoadVisitor.load(@stack_definition, configuration_update)
+          ForemanDeployments::Config::SaveVisitor.save(@stack_definition, @deployment.configuration)
+
+          @deployment.configuration.save!
+        end
+
+        api :GET, '/deployments/:id/', N_('Get information about a deployment')
+        param :id, :identifier, :required => true
+        def show
+          ForemanDeployments::Config::LoadVisitor.load(@stack_definition, @deployment.configuration)
+          # TODO: show deployment status (config, invalid, deploying, [reverting], deployed)
+          @validation_result = ForemanDeployments::Validation::Validator.validate(@stack_definition)
+        end
+
+        api :POST, '/deployments/:id/run/', N_('Start a deployment')
+        param :id, :identifier, :required => true
+        def run
+          # configure with user input
+          ForemanDeployments::Config::LoadVisitor.load(@stack_definition, @deployment.configuration)
+
+          # validate
+          result = ForemanDeployments::Validation::Validator.validate(@stack_definition)
+          fail "Validation failed:\n #{result.messages}" unless result.valid?
+
+          ForemanTasks.sync_task(Tasks::StackDeployAction, @stack_definition)
+        end
+
+        def parse_stack
```

Comment:
Let's move this method inside the `Deployment` class:
``` Ruby
def parsed_stack
  @stack_definition ||= begin
    definition = ForemanDeployments::StackParser.parse(self.stack.definition)
    ForemanDeployments::Config::LoadVisitor.load(definition, self.configuration)
    definition
  end
end
```
Motivation: Slim controllers, fat models.